homeassistant/components/nest/sensor.py

```diff
@@ -1,20 +1,104 @@
-"""Support for Nest sensors that dispatches between API versions."""
+"""Support for Google Nest SDM sensors."""
+from __future__ import annotations
 
+import logging
+
+from google_nest_sdm.device import Device
+from google_nest_sdm.device_manager import DeviceManager
+from google_nest_sdm.device_traits import HumidityTrait, TemperatureTrait
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTemperature
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import DATA_SDM
-from .legacy.sensor import async_setup_legacy_entry
-from .sensor_sdm import async_setup_sdm_entry
+from .const import DATA_DEVICE_MANAGER, DOMAIN
+from .device_info import NestDeviceInfo
+
+_LOGGER = logging.getLogger(__name__)
+
+
+DEVICE_TYPE_MAP = {
+    "sdm.devices.types.CAMERA": "Camera",
+    "sdm.devices.types.DISPLAY": "Display",
+    "sdm.devices.types.DOORBELL": "Doorbell",
+    "sdm.devices.types.THERMOSTAT": "Thermostat",
+}
 
 
 async def async_setup_entry(
```

Comment:
Reviewer note: This file is a rename of `sensor_sdm.py` with this method renamed.